doc/vim9.jax

```diff
@@ -537,30 +879,67 @@ Notes:
 	var dict = {'key with space': value}
 	var dict = {"key\twith\ttabs": value}
 	var dict = {'': value}  		# 空のキー
-
+<							*E1139*
 キーに式を使用する必要がある場合は、JavaScript と同様に角括弧を使用することが
 できます: >
 	var dict = {["key" .. nr]: value}
 
-キーの型には、文字列、数値、bool、floatのいずれかを指定できます。その他の型の
-場合はエラーが発生します。数字は、[] をつけてもつけなくても与えられます: >
-	var dict = {123: 'without', [456]: 'with'}
+キーの型には、文字列、数値、真偽値、浮動小数点のいずれかを指定できます。その他
+の型の場合はエラーが発生します。[] を使わない場合は文字列として扱われ、先頭の
+0 を保持します。[] を使って式が与えられた場合は、式を評価し、文字列に変換しま
+す。先頭の 0 は省かれます: >
+	var dict = {000123: 'without', [000456]: 'with'}
 	echo dict
-	{'456': 'with', '123': 'without'}
+	{'456': 'with', '000123': 'without'}
+[] 外ではドットは受け入れられないので、浮動小数点数は [] の中でのみ動作しま
+す: >
+	var dict = {[00.013]: 'float'}
+	echo dict
+	{'0.013': 'float'}
 
-:xit、:t、:k、:append、:change、:insert の廃止 ~
 
+:xit、:t、:k、:append、:change、:insert の廃止 ~
+							*E1100*
 これらのコマンドは容易にローカル変数の名前と混同します。
 `:x` や `:xit` の代わりに `:exit` を使用できます。
 `:t` の代わりに `:copy` を使用できます。
 `:k` の代わりに `:mark` を使用できます。
 
 比較 ~
 
-オプション 'ignorecase' は文字列の比較には作用しません。
+オプション 'ignorecase' は文字列の比較には作用しません。なので、"=~" は "=~#"
+と同じように動作します。
+
+現状では、"is" と "isnot" (|expr-is| と |expr-isnot|) を文字列に対して使うと
+常に偽を返します。旧来の Vim script では文字列の比較を行うだけでしたが、Vim9
+script ではそれらは文字列オブジェクトが同一であることをチェックします。文字列
+は使われるときに複製されるので、二つの文字列は同一と判定されることはありませ
+ん。 (この挙動はいつか文字列が複製されるのではなく、参照カウントで管理されるよ
+うになると変更されるかもしれません)
```

Comment:
「いつの日にか」だと詩的すぎるけど、ここで「いつか」としてしまうと誤解されやすそうでもありますね。

```suggestion
ん。 (この挙動については文字列が複製されるのではなく、参照カウントで管理するよ
うにいつか変更されるかもしれません)
```

とかかかな?

Reply:
確かに「いつか」としてしまうと、将来的に文字列管理の内部実装を変更する予定があるように誤解されやすそうですね。
ただ、
`this might change some day if strings are not copied but reference counted`
原文はこうで、「変更される("change")」という語の主語は "this" が指す「現状の Vim9 script では `is` を用いた比較は恒偽になる挙動」のことになると思うので、tsuyoshicho さんが提案してくださった案だと「変更される」の主語が「文字列管理の内部実装」になり、ちょっと原文から文意が離れすぎてしまうかな、と思います。
代わりに、原文の "if" を訳にも出して、
 - 「この挙動は、いつか、もし文字列が複製されるのではなく参照カウントで管理されるようになると、変更されるかもしれません」

としたり、あるいはもうちょっと語順を日本語として自然になりそうな感じにして
 - 「いつか、もし文字列が複製されるのではなく参照カウントで管理されるようになると、この挙動は変更されるかもしれません」

のような感じにするのはどうでしょうか。

Reply:
たしかにそうか、そのへんはおまかせします。

Reply:
後者の「いつか、もし文字列が複製されるのではなく参照カウントで管理されるようになると、この挙動は変更されるかもしれません」の訳に変更しました。